incubator/virtualcluster/pkg/syncer/conversion/mutate.go

```diff
@@ -46,67 +47,81 @@ func VC(mc *mc.MultiClusterController, clusterName string) VCMutateInterface {
 }
 
 func (m *mutator) Pod(pPod *v1.Pod) PodMutateInterface {
-	return &podMutator{mc: m.mc, clusterName: m.clusterName, pPod: pPod}
+	return &podMutateCtx{mc: m.mc, clusterName: m.clusterName, pPod: pPod}
 }
 
 func (m *mutator) Service(pService *v1.Service) ServiceMutateInterface {
 	return &serviceMutator{pService: pService}
 }
 
 type PodMutateInterface interface {
-	Mutate(pPod *v1.Pod, vSASecret, SASecret *v1.Secret, services []*v1.Service, nameServer string) error
+	Mutate(ms ...PodMutator) error
 }
 
-type podMutator struct {
+type PodMutator func(p *podMutateCtx) error
+
+type podMutateCtx struct {
 	mc          *mc.MultiClusterController
 	clusterName string
 	pPod        *v1.Pod
 }
 
-// getCluster returns the cluster name of the corresponding virtualcluster
-func (p *podMutator) getCluster() string {
-	ans := p.pPod.GetAnnotations()
-	return ans[constants.LabelCluster]
-}
-
 // MutatePod convert the meta data of containers to super master namespace.
 // replace the service account token volume mounts to super master side one.
-func (p *podMutator) Mutate(vPod *v1.Pod, vSASecret, SASecret *v1.Secret, services []*v1.Service, nameServer string) error {
-	p.pPod.Status = v1.PodStatus{}
-	p.pPod.Spec.NodeName = ""
+func (p *podMutateCtx) Mutate(ms ...PodMutator) error {
+	for _, mutator := range ms {
+		if err := mutator(p); err != nil {
+			return err
+		}
+	}
 
-	// setup env var map
-	serviceEnv := getServiceEnvVarMap(p.pPod.Namespace, p.getCluster(), *p.pPod.Spec.EnableServiceLinks, services)
+	return nil
+}
 
-	for i := range p.pPod.Spec.Containers {
-		mutateContainerEnv(&p.pPod.Spec.Containers[i], vPod, serviceEnv)
-		mutateContainerSecret(&p.pPod.Spec.Containers[i], vSASecret, SASecret)
-	}
+func PodMutateDefault(vPod *v1.Pod, vSASecret, SASecret *v1.Secret, services []*v1.Service, nameServer string) PodMutator {
```

Comment:
PodMutateDefault is the same as the old one. In addition, I mutate the hostAlias here